ui/desktop/electron-app/src/helpers/app-updater.js

```diff
@@ -19,6 +20,31 @@ if (debug && process.env.APP_UPDATER_CURRENT_VERSION) {
   currentVersion = process.env.APP_UPDATER_CURRENT_VERSION;
 }
 
+// Returns the real CPU architecture of the machine.
+const returnArchitectureToUpdate = () => {
+  const nodeArchitecture = process.arch;
+
+  if (nodeArchitecture === 'arm64') {
+    return nodeArchitecture;
+  }
+
+  try {
+    const runsOnRosetta = execSync(
+      'sysctl -in sysctl.proc_translated',
+    ).toString();
+
+    if (runsOnRosetta) {
+      return 'arm64';
+    } else {
+      return nodeArchitecture; // amd64
+    }
+  } catch (err) {
+    console.error(`returnArchitectureToUpdate error: ${err}`);
+  }
+
+  return 'arm64';
```

Comment:
This only gets here if we get an error right? Should we just return back `nodeArchitecture` to be safe and remove line 39 since we don't need that else then?